### ISSUE TITLE
Some styles are being returned on a new line using streaming

### ISSUE THREAD

**Issue by hakimelek** [styled-components/styled-components#1478]
## Version
styled-components: 3.1.5

## Reproduction

I am trying to use the new streaming API. I am using `redux`, `react-router` and `react-loadable`. This is how I have it setup so far on my server file:

```js

    const sheet = new ServerStyleSheet()
    let modules = []

    const stream = sheet.interleaveWithNodeStream(renderToNodeStream(
      sheet.collectStyles(
        <Loadable.Capture report={moduleName => modules.push(moduleName)}>
          <Provider store={store}>
            <StaticRouter
              location={req.url}
              context={context}
              basename={basename}
            >
              <ThemeProvider>
                <App />
              </ThemeProvider>
            </StaticRouter>
          </Provider>
        </Loadable.Capture>
      )
    ))

    stream.pipe(res, { end: false })

    let bundles = getBundles(loadableStats, modules)

    const remainingChunk = html.remainingChunk(
      bundleName,
      bundles
    )
    
    stream.on('end', () => res.end(remainingChunk))
```

client file:
```js
const render = () => {
  consolidateStreamedStyles()
  Loadable.preloadReady().then(() => {
    ReactDOM.hydrate(
      <AppContainer>
        <App />
      </AppContainer>,
      document.getElementById('root')
    )
  })
}

render()

if (module.hot) {
  module.hot.accept('components/App', () => {
    render()
  })
}
```

The streaming seems to work fine except some of the characters are on a new line. Here is how the styles look like on the source view.

```
r
--
  | d
  | e
  | r
  | -
  | b
  | o
  | x
  | ;
  | }
  | /* sc-component-id: sc-keyframes-gPFNxc */
  | @-webkit-keyframes gPFNxc{0%{top:12px;-webkit-transform:scale(0);-ms-transform:scale(0);transform:scale(0);}40%{-webkit-transform:scale(1.0);-ms-transform:scale(1.0);transform:scale(1.0);}50%{top:-12px;}60%{-webkit-transform:scale(1.0);-ms-transform:scale(1.0);transform:scale(1.0);}100%{top:12px;-webkit-transform:scale(0);-ms-transform:scale(0);transform:scale(0);}}
  | @keyframes gPFNxc{0%{top:12px;-webkit-transform:scale(0);-ms-transform:scale(0);transform:scale(0);}40%{-webkit-transform:scale(1.0);-ms-transform:scale(1.0);transform:scale(1.0);}50%{top:-12px;}60%{-webkit-transform:scale(1.0);-ms-transform:scale(1.0);transform:scale(1.0);}100%{top:12px;-webkit-transform:scale(0);-ms-transform:scale(0);transform:scale(0);}}
  | /* sc-component-id: sc-bwzfXH */
  | .sc-bwzfXH {}
  |  
  |  
  | .
  | e
  | U
  | h
  | P
  | Z
  | g
  | {
  | p
  | a
  | d
  | d
  | i
  | n
  | g
  | -
  | l
  | e
  | f
  | t
  | :

```

Any ideas?

**Comment by probablyup** [comment]
Interesting. Would you happen to have a reproduction repo I could take a look at?

**Comment by probablyup** [comment]
I think the order of your code should be a little different:

```js
    stream.pipe(res, { end: false })
    stream.on('end', () => {
        let bundles = getBundles(loadableStats, modules)

        const remainingChunk = html.remainingChunk(
            bundleName,
            bundles
        )
            
        res.write(remainingChunk)
        res.end();
    })
```

**Comment by hakimelek** [comment]
Thought that would solve it as well. I tried it and I got an `Error: write after end` error. I am getting the same behavior when I uncomment the bundle and remainingChunk part. Let me try to setup a small project and see if I get same the behavior.

**Comment by hakimelek** [comment]
It seems that the problem is in this line: https://github.com/styled-components/styled-components/blob/76d7c693d8d4e21b522ad9906c21d55214e6ac4b/src/models/ServerStyleSheet.js#L74

I didn't totally understand what were you trying to achieve here, but changing the code to something that looks like this, solved the problem:

```js
    for (var i = 0; i < cssRulesSize; i += 1) {
      var cssRule = css[i];
      comp.css += cssRule;
    }
    comp.css += '\n';
```

Also, it's good to note that I am server side rendering on Linux CentOS, that would interpret line breaking differently.

**Comment by probablyup** [comment]
Hmm I think it's meant to coalesce multiple newlines into a single one. Want to make a PR?

**Comment by hakimelek** [comment]
Sure!

**Comment by probablyup** [comment]
Transitively fixed in https://github.com/styled-components/styled-components/pull/1501. It'll be released with `v3.2.0` soon.
